app/presenters/single_content_item_presenter.rb

```diff
@@ -213,16 +213,42 @@ def calculate_current_pageviews_per_visit
   end
 
   def calculate_previous_pageviews_per_visit
-    previous = if @previous_metrics['pviews'][:value].to_f.zero? || @previous_metrics['upviews'][:value].to_f.zero?
+    previous = if @previous_metrics['pviews'][:value].blank? || @previous_metrics['upviews'][:value].blank?
```

Comment:
Great to see the changes come into shape. One last thing please: the `single_content_item_presenter.rb` we've now got more calculation methods hidden in a private method. The other private methods in here were relatively low complexity. Now that we have methods that have increased complexity, it looks like a good time to abstract them into a new class that does calculations. Basically the `calculate_` methods look like good candidates for this abstraction. We can also test those in isolation.